### PR TITLE
Drop support for EOL Python 2.6 and 3.3

### PR DESCRIPTION
**Pull request by hugovk**
Python 2.6 and 3.3 are EOL (since 2013-10-29 and 2017-09-29 respectively) and no longer receiving security updates (or any updates) from the core Python team.

They're also little used.

Here's the pip installs for toolz from PyPI for August 2018:

| python_version | percent | download_count |
| -------------- | ------: | -------------: |
| 3.6            |  87.55% |      3,618,284 |
| 2.7            |   7.55% |        312,216 |
| 3.5            |   3.88% |        160,493 |
| 3.4            |   0.56% |         23,167 |
| 3.7            |   0.45% |         18,596 |
| 3.8            |   0.00% |             44 |
| 3.3            |   0.00% |             19 |
| 2.6            |   0.00% |              4 |
| Total          |         |      4,132,823 |

Source: `pypinfo --start-date 2018-08-01 --end-date 2018-08-31 --percent --markdown toolz pyversion`

Removing them allows redundant code to be removed and more modern features of Python to be used.

### CODE REVIEW

**Comment by groutr** [comment]
@hugovk Switching from % formatting to `.format` typically has performance implications.  I'm against doing that without some sort of performance testing.

**Comment by hugovk** [comment]
Timing just `%` v. `.format()`:

```console
$ python -mtimeit -s"modname, qualname = 'name1', 'name2'" "func = '%s:%s' % (modname, qualname)"
1000000 loops, best of 5: 243 nsec per loop
$ python -mtimeit -s"modname, qualname = 'name1', 'name2'" "func = '{}:{}'.format(modname, qualname)"
1000000 loops, best of 5: 367 nsec per loop
```

`%` formatting is quicker by about a third. I doubt that's going to make much difference in real life.

Running pytest on this branch (third run, to ensure no warmup costs):


```
platform darwin -- Python 3.7.0, pytest-3.8.1, py-1.5.4, pluggy-0.7.1
...
====================================== 199 passed, 1 warnings in 8.30 seconds ======================================
```

Running pytest on branch [rm-eol2](https://github.com/hugovk/toolz/tree/rm-eol2), same as this branch but keeps `%` formatting (third run):

```
====================================== 199 passed, 1 warnings in 8.31 seconds ======================================
```

Show no real difference. So this may be premature optimisation. 

I find `.format()` a bit clearer to read and use but others prefer the older C style formatting, so I'll leave it down to your personal choice -- here's a PR that retains it (https://github.com/pytoolz/toolz/pull/422) so you can take your pick :)

File: toolz/functoolz.py [hunk 1]
```diff
@@ -344,14 +345,14 @@ def __reduce__(self):
             if isinstance(obj, curry) and obj.func is func:
                 is_decorated = obj is self
                 qualname = '.'.join(attrs)
-                func = '%s:%s' % (modname, qualname)
+                func = '{}:{}'.format(modname, qualname)
```
[the remark this answers is not in the document]
Comment (hugovk) [reply toolz/functoolz.py]: Updated!

**Comment by eriknw** [comment]
Closed in favor of #422 .  Thanks for the PR and discussion all.

**Pull request closed without merging**